Detail required settings in Gateway doc (#6126)
hashicorp/consul: website/source/docs/connect/mesh_gateway.html.md

2. General network connectivity to all services within its local Consul datacenter.
3. General network connectivity to all mesh gateways within remote Consul datacenters.

> Mesh gateways also requre that your Consul datacenters are configured correctly:

Reviewer: ```suggestion
Mesh gateways also require that your Consul datacenters are configured correctly:
```

> - You'll need to use Consul version 1.6.0beta1.

Reviewer: ```suggestion
- You'll need to use Consul version 1.6.0-beta1.
```
https://releases.hashicorp.com/consul/1.6.0-beta1/

> - Consul [Connect](http://localhost:4567/docs/agent/options.html#connect) must be enabled in both datacenters.

Reviewer: ```suggestion
- Consul [Connect](/docs/agent/options.html#connect) must be enabled in both datacenters.
```

> - Each of your [datacenters](/docs/agent/options.html#datacenter) must have a unique name. 

Author: ```suggestion
- Each of your [datacenters](/docs/agent/options.html#datacenter) must have a unique name. 
- Your datacenters must be [WAN joined](https://learn.hashicorp.com/consul/security-networking/datacenters).
```

> - The [primary datacenter](/docs/agent/options.html#primary_datacenter) must be set to the same value in both datacenters.

Reviewer: ```suggestion
- The [primary datacenter](/docs/agent/options.html#primary_datacenter) must be set to the same value in both datacenters. This specifies which datacenter is the authority for Connect certificates and is required for all datacenters to trust requests from each other.
```

Might be worth explaining a bit here?

> - [gRPC](http://localhost:4567/docs/agent/options.html#grpc_port) must be enabled. 

Reviewer: ```suggestion
- [gRPC](/docs/agent/options.html#grpc_port) must be enabled. 
```

> - If you want to set the default gateway mode you must enable [centralized configuration](/docs/agent/options.html#config_entries). 

Reviewer: ```suggestion
- If you want to [enable gateways globally](/docs/connect/mesh_gateway.html#enabling-gateways-globally) you must enable [centralized configuration](/docs/agent/options.html#enable_central_service_config). 
```

I think that the option is actually https://www.consul.io/docs/agent/options.html#enable_central_service_config

2. General network connectivity to all services within its local Consul datacenter.
3. General network connectivity to all mesh gateways within remote Consul datacenters.

Mesh gateways also require that your Consul datacenters are configured correctly:

> - You'll need to use Consul version 1.6.0-beta1.

Reviewer 2: beta2 will work too. Should we just say 1.6.0 here maybe? Not a big deal regardless but if we say 1.6.0 then we wont need to update this again and I think its specific enough that someone would know that they can use 1.6.0 beta.

Author: ```suggestion
- You'll need to use Consul version 1.6.0.
```

- Consul [Connect](/docs/agent/options.html#connect) must be enabled in both datacenters.
- Each of your [datacenters](/docs/agent/options.html#datacenter) must have a unique name. 
- Your datacenters must be [WAN joined](https://learn.hashicorp.com/consul/security-networking/datacenters).

> - The [primary datacenter](/docs/agent/options.html#primary_datacenter) must be set to the same value in both datacenters. This specifies which datacenter is the authority for Connect certificates and is required for all datacenters to trust requests from each other.

Reviewer: ```suggestion
- The [primary datacenter](/docs/agent/options.html#primary_datacenter) must be set to the same value in both datacenters. This specifies which datacenter is the authority for Connect certificates and is required for all datacenters to trust Connect requests from each other.
```

Maybe it's worth being clear that this is only needed for Connect requests. RPC would still work for example.

Author: Would it be accurate to say 

"and is required for services in all datacenters to establish mutual TLS with each other?" 

"Connect" is a bit overloaded and so I think it might be better to be more specific :)

Reviewer: Yeah I think that is good!
